backend/app/services/failed_price_retry_runner.py

```
@@ -0,0 +1,147 @@
+                price_cache=price_cache,
+                db=None,
+                jobs=(retry_job,),
+                total=len(symbols),
+                batch_size=max(1, int(batch_size)),
+                market=market,
+                market_for_symbol=lambda _symbol: market,
+                after_batch=after_batch,
+            )
+        except PriceRefreshBatchExecutionError as exc:
+            if isinstance(exc.cause, SoftTimeLimitExceeded):
+                raise exc.cause
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)